include/picongpu/simulationControl/MySimulation.hpp

```diff
@@ -213,16 +214,26 @@ class MySimulation : public SimulationHelper<simDim>
 
             // calculate local grid points & offset
             gridSizeLocal[dim] = parserGD.getLocalSize(myGPUpos[dim]);
-            gridOffset[dim] = parserGD.getOffset(myGPUpos[dim], global_grid_size[dim]);
         }
         // by default: use an equal distributed box for all omitted params
         for (uint32_t dim = gridDistribution.size(); dim < simDim; ++dim)
         {
-            gridSizeLocal[dim] = global_grid_size[dim] / gpus[dim];
-            gridOffset[dim] = gridSizeLocal[dim] * myGPUpos[dim];
+            gridSizeLocal[dim] = globalGridSize[dim] / gpus[dim];
         }
 
-        Environment<simDim>::get().initGrids(global_grid_size, gridSizeLocal, gridOffset);
+        AdjustDomainSize adjustDomainSize(
+            gpus,
+            myGPUpos,
+            isPeriodic,
+            slidingWindow
+        );
+
+        DataSpace<simDim> gridOffset;
+
+        // adjust domain offset and sizes
```

Comment:
I think the next line very clearly explains itself and so the comment is not needed.